frontend/public/components/catalog.jsx

```diff
@@ -0,0 +1,209 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import * as fuzzy from 'fuzzysearch';
+import * as PropTypes from 'prop-types';
+import { Helmet } from 'react-helmet';
+
+import { FLAGS, connectToFlags, flagPending } from '../features';
+import { Firehose, NavTitle, StatusBox } from './utils';
+import { CatalogList } from './catalog-items';
+import { getBuilderSpecTagName, getImageStreamTagAnnotation } from './image-stream';
+import { serviceClassDisplayName } from '../module/k8s';
+import { normalizeIconClass, ClusterServiceClassIcon } from './cluster-service-class-icon';
+
+class CatalogListPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleFilterChange = this.handleFilterChange.bind(this);
+    this.clearFilter = this.clearFilter.bind(this);
+
+    this.state = {
+      filterValue: '',
+      items: [],
+      filteredItems: [],
+    };
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const {clusterserviceclasses, namespace } = this.props;
+    const {filterValue} = this.state;
+    if (namespace !== prevProps.namespace || clusterserviceclasses !== prevProps.clusterserviceclasses) {
+      this.createCatalogData();
+    } else if (filterValue !== prevState.filterValue) {
+      const filteredItems = this.filterItems(this.state.items);
+      this.setState({
+        filteredItems
+      });
+    }
+  }
+
+  filterItems(items) {
+    const {filterValue} = this.state;
+
+    if (!filterValue) {
+      return items;
+    }
+
+    const filterString = filterValue.toLowerCase();
+    return _.filter(items, item => {
+      return fuzzy(filterString, _.get(item, 'metadata.name', ''));
```

Comment:
We should be filtering display names, not names.

Reply:
Why doesn't the comment above show up under 'Conversation'?  I didn't see these until I looked under 'Files Changed'.

There is a whole filter panel that is going to be used for the Catalog, this PR is primarily for just getting the data right.  I'd like to defer all sorting and filtering to a later PR.  -thanks

Reply:
> There is a whole filter panel that is going to be used for the Catalog, this PR is primarily for just getting the data right. I'd like to defer all sorting and filtering to a later PR. -thanks

Then we should remove filtering code from this PR. Otherwise there is a danger we ship an incorrect implementation if we don't finish the filter panel before feature freeze.